source/api/app.py

```diff
@@ -246,6 +248,52 @@ def format_operation_response(result, error_message):
     return response
 
 
+def test_nfs_access(group, mount_target_ip):
+    rules = EC2.describe_security_group_rules(
+        Filters=[
+                {
+                    'Name': 'group-id',
+                    'Values': [group]
+                }
+            ]
+        )
+
+    contains_valid_rule = False
+
+    for rule in rules['SecurityGroupRules']:
+        # check if rule is inbound
+        if rule['IsEgress'] is False:
+            if 'ReferencedGroupInfo' in rule:
+                # references a sg in the rule
+                ref_group_id = rule['ReferencedGroupInfo']['GroupId']
+                if ref_group_id == rule['GroupId']:
+                    if rule['IpProtocol'] == '-1' or rule['IpProtocol'] == 'tcp':
+                        if rule['FromPort'] and rule['ToPort'] == 2049 or rule['FromPort'] and rule['ToPort'] == -1:
```

Comment:
does python have an order of operations with this sort of thing?

Reply:
Yes, here is a good description of how python evaluates and / or expressions:

> The expression x and y first evaluates x; if x is false, its value is returned; otherwise, y is evaluated and the resulting value is returned.
> 
> The expression x or y first evaluates x; if x is true, its value is returned; otherwise, y is evaluated and the resulting value is returned.